qiskit/pulse/ir/ir.py

```diff
@@ -217,11 +222,19 @@ def _draw_nodes(n):
     def flatten(self, inplace: bool = False) -> SequenceIR:
         """Recursively flatten the SequenceIR.
 
+        The flattening process includes breaking up nested IRs until only instructions remain.
```

Comment:
Can we provide this functionality as a pass? Seems like no existing pass requires flatten. I'm okey with visualizing a graph with nested (unflatten) sequences, if this is only for visualization purpose. Some target might require flatten if it doesn't have notion of context, so having the flatten pass still makes sense to me.

Reply:
For the visualization purpose we can add a dedicated visualization function to the visualization module, and call the flatten pass inside depending on the argument, e.g. 
```
def visualize_sequence_ir(sequence_ir: SequenceIR, flat_nested: bool = True):
```

Reply:
That makes sense as well. I will split flatten out in a separate PR. (It wasn't introduced in this PR - only its tests)